### PR TITLE
New server management module

### PR DESCRIPTION
**Pull request by t-woerner**
There is a new server management module placed in the plugins folder:

    plugins/modules/ipaserver.py

The server module allows to ensure presence and absence of servers. The
module requires an existing server, the deployment of a new server can
not be done with the module.

Here is the documentation for the module:

    README-server.md

New example playbooks have been added:

    playbooks/server/server-absent-continue.yml
    playbooks/server/server-absent-force.yml
    playbooks/server/server-absent-ignore_last_of_role.yml
    playbooks/server/server-absent-ignore_topology_disconnect.yml
    playbooks/server/server-absent.yml
    playbooks/server/server-hidden.yml
    playbooks/server/server-location.yml
    playbooks/server/server-no-location.yml
    playbooks/server/server-no-service-weight.yml
    playbooks/server/server-not-hidden.yml
    playbooks/server/server-present.yml
    playbooks/server/server-service-weight.yml

New tests for the module:

    tests/server/test_server.yml

### CODE REVIEW

File: tests/server/test_server.yml [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+- name: Test server
+  hosts: ipaserver
+  become: true
+
+  tasks:
+
+  # CLEANUP TEST ITEMS
+
+  - name: Ensure server "{{ 'ipaserver.' + ipaserver_domain }}" without location
+    ipaserver:
```
Comment (rjeffman) [inline tests/server/test_server.yml:11]: Please, add `ipaadmin_password` to all the tests.

Comment (t-woerner) [reply tests/server/test_server.yml:11]: Done

File: README-server.md [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+
+  tasks:
+  - ipaserver:
+      ipaadmin_password: SomeADMINpassword
+      name: server.example.com
+      service_weight: -1
+```
+
+
+Example playbook to make sure server "server.example.com" is present as hidden:
```
[the remark this answers is not in the document]
Comment (t-woerner) [reply README-server.md]: Done, also in ipaservice.py

File: README-server.md [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+
+  tasks:
+  - ipaserver:
+      ipaadmin_password: SomeADMINpassword
+      name: server.example.com
+      hidden: yes
+```
+
+
+Example playbook to make sure server "server.example.com" is present as not hidden:
```
[the remark this answers is not in the document]
Comment (t-woerner) [reply README-server.md]: Done, also in ipaservice.py

**Review by rjeffman — APPROVED** [review]
LGTM.

**Pull request merged**